src/reactions/edit.js

```diff
@@ -36,20 +39,102 @@ const DUMMY_REACTIONS = {
 		),
 		items: generateReactionItems( 6, 'Reposter', 82, COLORS ), // 82 is ASCII for 'R'
 	},
+	quotes: {
+		label: sprintf(
+			/* translators: %d: Number of quotes */
+			_x( '%d quotes', 'number of quotes', 'activitypub' ),
+			7
+		),
+		items: generateReactionItems( 7, 'Quoter', 81, COLORS ), // 81 is ASCII for 'Q'
+	},
+};
+
+// Dummy counts for summary view.
+const DUMMY_COUNTS = {
+	comments: 5,
+	likes: 9,
+	reposts: 6,
+	quotes: 0,
 };
 
+/**
+ * Summary component for displaying reaction counts.
+ *
+ * @param {Object} props           Component props.
+ * @param {string} props.separator Separator between items.
+ * @return {JSX.Element} Component to render.
+ */
+function Summary( { separator } ) {
+	const items = [
+		{ label: __( 'Comments', 'activitypub' ), count: DUMMY_COUNTS.comments },
+		{ label: __( 'Likes', 'activitypub' ), count: DUMMY_COUNTS.likes },
+		{ label: __( 'Reposts', 'activitypub' ), count: DUMMY_COUNTS.reposts },
+		{ label: __( 'Quotes', 'activitypub' ), count: DUMMY_COUNTS.quotes },
+	];
+
+	return (
+		<div className="activitypub-reactions-summary">
+			{ items.map( ( item, index ) => (
+				<span key={ item.label }>
+					{ index > 0 && <span className="reactions-summary-separator">{ separator }</span> }
+					<span className="reactions-summary-item">
+						<span className="reactions-summary-count">{ item.count }</span>
+						<span className="reactions-summary-label">{ item.label }</span>
+					</span>
+				</span>
+			) ) }
+		</div>
+	);
+}
+
 /**
  * Edit component for the Reactions block.
  *
- * @param {Object} props                            Block props.
- * @param          props.__unstableLayoutClassNames Layout class names.
+ * @param {Object} props                              Block props.
+ * @param {Object} props.attributes                   Block attributes.
+ * @param {string} props.__unstableLayoutClassNames   Layout class names.
  * @return {JSX.Element} Component to render.
  */
-export default function Edit( { __unstableLayoutClassNames } ) {
+export default function Edit( { attributes, setAttributes, __unstableLayoutClassNames } ) {
+	const {
+		className = '',
+		displayStyle = 'facepile',
+		separator = ', ',
+		showComments = true,
+		showEmpty = true,
+	} = attributes;
 	const blockProps = useBlockProps( {
 		className: __unstableLayoutClassNames,
 	} );
 	const { getCurrentPostId } = select( 'core/editor' );
+	const { showAvatars = true } = useOptions();
+	const hasInitialized = useRef( false );
+
+	// On first render, if avatars are disabled and no style class is set, default to summary.
+	useEffect( () => {
+		if ( hasInitialized.current ) {
+			return;
+		}
+		hasInitialized.current = true;
+
+		// Only apply default if no style has been explicitly chosen yet.
+		const hasStyleClass = className?.includes( 'is-style-' );
+		if ( ! showAvatars && ! hasStyleClass ) {
+			setAttributes( {
+				className: ( className ? className + ' ' : '' ) + 'is-style-summary',
```

Comment:
We add `clsx` as a dependency in `add/router`, maybe we could move that up into this PR and use it here.